versions/draft-2/parsers/antlr4/WdlDraft2Parser.g4

```diff
@@ -0,0 +1,268 @@
+parser grammar WdlDraft2Parser;
+
+options { tokenVocab=WdlLexer; }
+
+map_type
+	: MAP LBRACK wdl_type COMMA wdl_type RBRACK
+	;
+
+array_type
+	: ARRAY LBRACK wdl_type RBRACK PLUS?
+	;
+
+pair_type
+	: PAIR LBRACK wdl_type COMMA wdl_type RBRACK
+	;
+
+type_base
+	: array_type
+	| map_type
+	| pair_type
+	| (STRING | FILE | BOOLEAN | OBJECT | INT | FLOAT)
+	;
+
+wdl_type
+	: (type_base OPTIONAL | type_base)
+	;
+
+unbound_decls
+	: wdl_type Identifier
+	;
+
+bound_decls
+	: wdl_type Identifier EQUAL expr
+	;
+
+any_decls
+	: unbound_decls
+	| bound_decls
+	;
+
+number
+	: IntLiteral
+	| FloatLiteral
+	;
+
+expression_placeholder_option
+	: BoolLiteral EQUAL (string | number)
+	| DEFAULT EQUAL (string | number)
+	| SEP EQUAL (string | number)
+	;
+
+string_part
+	: StringPart*
+	;
+
+string_expr_part
+	: StringCommandStart (expression_placeholder_option)* expr RBRACE
+	;
+
+string_expr_with_string_part
+	: string_expr_part string_part
+	;
+
+string
+	: DQUOTE string_part string_expr_with_string_part* DQUOTE
+	| SQUOTE string_part string_expr_with_string_part* SQUOTE
+	;
+
+primitive_literal
+	: BoolLiteral
+	| number
+	| string
+	| Identifier
+	;
+
+expr
+	: expr_infix
+	;
+
+expr_infix
+	: expr_infix0 #infix0
+	;
+
+expr_infix0
+	: expr_infix0 OR expr_infix1 #lor
+	| expr_infix1 #infix1
+	;
+
+expr_infix1
+	: expr_infix1 AND expr_infix2 #land
+	| expr_infix2 #infix2
+	;
+
+expr_infix2
+	: expr_infix2 EQUALITY expr_infix3 #eqeq
+	| expr_infix2 NOTEQUAL expr_infix3 #neq
+	| expr_infix2 LTE expr_infix3 #lte
+	| expr_infix2 GTE expr_infix3 #gte
+	| expr_infix2 LT expr_infix3 #lt
+	| expr_infix2 GT expr_infix3 #gt
+	| expr_infix3 #infix3
+	;
+
+expr_infix3
+	: expr_infix3 PLUS expr_infix4 #add
+	| expr_infix3 MINUS expr_infix4 #sub
+	| expr_infix4 #infix4
+	;
+
+expr_infix4
+	: expr_infix4 STAR expr_infix5 #mul
+	| expr_infix4 DIVIDE expr_infix5 #divide
+	| expr_infix4 MOD expr_infix5 #mod
+	| expr_infix5 #infix5
+	;
+
+expr_infix5
+	: expr_core
+	;
+
+expr_core
```

Comment:
See https://github.com/openwdl/wdl/pull/364 for an ordering issue that @orodeh reported